# CONSULTAS API
Para optimizar el proceso de consultas y teniendo en cuenta que esto es un PMV (Producto Mínimo Viable), he decido realizar ciertas transformaciones adicionales sobre los datasets anteriormente trabajados, de manera que el despliegue sea fácil y eficiente. 

#### Importar librerías

In [40]:
import pandas as pd
import ast

#### Cargar el conjunto de datos limpios

In [41]:
df_steam_games = pd.read_csv(r'C:\PI1\data\DataLimpia\steam_games_cln.csv')
df_user_reviews = pd.read_csv(r'C:\PI1\data\DataLimpia\user_reviews_cln.csv')
df_user_items = pd.read_csv(r'C:\PI1\data\DataLimpia\users_items_cln.csv')


In [ ]:
#La columna year tenía valores como flotantes y los converti en enteros
df_steam_games["year"] = df_steam_games["year"].astype('Int64')
df_user_reviews["year_posted"] = df_user_reviews["year_posted"].astype("Int64")

In [ ]:
df_user_items

item_id                      item_name            user_id  hours_game
0             10                 Counter-Strike  76561197970982479        0.10
1             30                  Day of Defeat  76561197970982479        0.12
2            300          Day of Defeat: Source  76561197970982479       78.88
3            240         Counter-Strike: Source  76561197970982479       30.88
4           3830                    Psychonauts  76561197970982479        5.55
...          ...                            ...                ...         ...
3246370   304930                       Unturned  76561198329548331       11.28
3246371   227940              Heroes & Generals  76561198329548331        0.72
3246372   388490  One Way To Die: Steam Edition  76561198329548331        0.05
3246373   521570          You Have 10 Seconds 2  76561198329548331        0.07
3246374   519140                     Minds Eyes  76561198329548331        0.05

[3246375 rows x 4 columns]

In [ ]:
df_steam_games

genres  \
0      ['Action', 'Casual', 'Indie', 'Simulation', 'S...   
1           ['Free to Play', 'Indie', 'RPG', 'Strategy']   
2      ['Casual', 'Free to Play', 'Indie', 'Simulatio...   
3                      ['Action', 'Adventure', 'Casual']   
4                  ['Action', 'Adventure', 'Simulation']   
...                                                  ...   
22516         ['Action', 'Adventure', 'Casual', 'Indie']   
22517      ['Casual', 'Indie', 'Simulation', 'Strategy']   
22518                    ['Casual', 'Indie', 'Strategy']   
22519                  ['Indie', 'Racing', 'Simulation']   
22520                                ['Casual', 'Indie']   

                          title  \
0           Lost Summoner Kitty   
1                     Ironbound   
2       Real Pool 3D - Poolians   
3                       弹炸人2222   
4         Battle Royale Trainer   
...                         ...   
22516              Kebab it Up!   
22517            Colony On Mars   
22518  LOGistICAL: South Africa   
22519             Russian Roads   
22520       EXIT 2 - Directions   

                                                    tags  \
0      ['Strategy', 'Action', 'Indie', 'Casual', 'Sim...   
1      ['Free to Play', 'Strategy', 'Indie', 'RPG', '...   
2      ['Free to Play', 'Simulation', 'Sports', 'Casu...   
3                      ['Action', 'Adventure', 'Casual']   
4      ['Action', 'Adventure', 'Simulation', 'FPS', '...   
...                                                  ...   
22516  ['Action', 'Indie', 'Casual', 'Violent', 'Adve...   
22517      ['Strategy', 'Indie', 'Casual', 'Simulation']   
22518                    ['Strategy', 'Indie', 'Casual']   
22519                  ['Indie', 'Simulation', 'Racing']   
22520  ['Indie', 'Casual', 'Puzzle', 'Singleplayer', ...   

                                                   specs  price  item_id  \
0                                      ['Single-player']   4.99   761140   
1      ['Single-player', 'Multi-player', 'Online Mult...   0.00   643980   
2      ['Single-player', 'Multi-player', 'Online Mult...   0.00   670290   
3                                      ['Single-player']   0.99   767400   
4                ['Single-player', 'Steam Achievements']   3.99   772540   
...                                                  ...    ...      ...   
22516  ['Single-player', 'Steam Achievements', 'Steam...   1.99   745400   
22517            ['Single-player', 'Steam Achievements']   1.99   773640   
22518  ['Single-player', 'Steam Achievements', 'Steam...   4.99   733530   
22519  ['Single-player', 'Steam Achievements', 'Steam...   1.99   610660   
22520  ['Single-player', 'Steam Achievements', 'Steam...   4.99   658870   

                      developer  year  
0                     Kotoshiro  2018  
1              Secret Level SRL  2018  
2                  Poolians.com  2017  
3                          彼岸领域  2017  
4           Trickjump Games Ltd  2018  
...                         ...   ...  
22516           Bidoniera Games  2018  
22517        Nikita "Ghost_RUS"  2018  
22518                    Sacada  2018  
22519  Laush Dmitriy Sergeevich  2018  
22520             xropi,stev3ns  2017  

[22521 rows x 8 columns]

## CONSULTAS

### 1. PlayTimeGenre
Debe devolver año con mas horas jugadas para dicho género.

Ejemplo de retorno: {"Año de lanzamiento con más horas jugadas para Género X" : 2013}

In [ ]:
# INNER JOIN de los DataFrames
join_items_games = pd.merge(df_user_items, df_steam_games[['item_id', 'year', 'genres']], on='item_id', how='inner')

In [ ]:
join_items_games

item_id                      item_name            user_id  \
0             10                 Counter-Strike  76561197970982479   
1             30                  Day of Defeat  76561197970982479   
2            300          Day of Defeat: Source  76561197970982479   
3            240         Counter-Strike: Source  76561197970982479   
4           3830                    Psychonauts  76561197970982479   
...          ...                            ...                ...   
2676077   417860                  Emily is Away  76561198323066619   
2676078   304930                       Unturned  76561198329548331   
2676079   227940              Heroes & Generals  76561198329548331   
2676080   388490  One Way To Die: Steam Edition  76561198329548331   
2676081   521570          You Have 10 Seconds 2  76561198329548331   

         hours_game  year                                             genres  
0              0.10  2000                                         ['Action']  
1              0.12  2003                                         ['Action']  
2             78.88  2010                                         ['Action']  
3             30.88  2004                                         ['Action']  
4              5.55  2005                                         ['Action']  
...             ...   ...                                                ...  
2676077        0.13  2015  ['Adventure', 'Casual', 'Free to Play', 'Indie...  
2676078       11.28  2017  ['Action', 'Adventure', 'Casual', 'Free to Pla...  
2676079        0.72  2016  ['Action', 'Free to Play', 'Indie', 'Massively...  
2676080        0.05  2015                      ['Adventure', 'Free to Play']  
2676081        0.07  2016                ['Casual', 'Free to Play', 'Indie']  

[2676082 rows x 6 columns]

In [ ]:
#Agrupamos por genero y año de lanzamiento
join_items_games = join_reviews_steam.groupby(['year', 'genres'])['hours_game'].sum().reset_index()

In [ ]:
join_items_games

year        genres  hours_game
0    1983        Action       57.92
1    1983     Adventure       57.92
2    1983        Casual       57.92
3    1984        Action        6.40
4    1984     Adventure        6.40
..    ...           ...         ...
315  2018        Action        0.35
316  2018     Adventure        0.35
317  2018  Early Access       65.41
318  2018         Indie        2.07
319  2018           RPG       63.69

[320 rows x 3 columns]

In [ ]:
join_items_games.dtypes

year            Int64
genres         object
hours_game    float64
dtype: object

In [ ]:
#Convertimos la columna genres a tipo lista
#La función ast.literal_eval evalúa la expresión de Python contenida en una cadena y devuelve el objeto correspondiente en Python. 
join_items_games['genres'] = join_items_games['genres'].apply(ast.literal_eval)

In [ ]:
join_items_games

year        genres  hours_game
0    1983        Action       57.92
1    1983     Adventure       57.92
2    1983        Casual       57.92
3    1984        Action        6.40
4    1984     Adventure        6.40
..    ...           ...         ...
315  2018        Action        0.35
316  2018     Adventure        0.35
317  2018  Early Access       65.41
318  2018         Indie        2.07
319  2018           RPG       63.69

[320 rows x 3 columns]

In [ ]:
#este código fragmenta una columna de listas en varias filas para hacer que los datos sean más accesibles y manejables.
join_items_games = join_reviews_steam.explode('genres').reset_index(drop=True)
join_items_games

year        genres  hours_game
0     1983        Action       57.92
1     1983     Adventure       57.92
2     1983        Casual       57.92
3     1984        Action        6.40
4     1984     Adventure        6.40
...    ...           ...         ...
5272  2018  Early Access        0.35
5273  2018         Indie        1.72
5274  2018  Early Access        1.72
5275  2018           RPG       63.34
5276  2018  Early Access       63.34

[5277 rows x 3 columns]

In [ ]:
# este código encuentra el juego más jugado (basado en las horas de juego) para cada género en cada año.
idxmax_genre = join_reviews_steam.loc[join_reviews_steam.groupby(['year', 'genres'])['hours_game'].idxmax()]

In [ ]:
# Ordenar el DataFrame resultante
idxmax_genre = idxmax_genre.sort_values(by=['genres', 'hours_game', 'year'], ascending=[True, False, True])

# Guardar el resultado en un archivo CSV
#idxmax_genre.to_csv('data/DataAPI/PlayTimeGenre.csv', index=False)

In [ ]:
idxmax_genre

year          genres   hours_game
1273  2012          Action  13212007.97
949   2011          Action   2822710.67
723   2009          Action   2383607.40
328   2004          Action   2160288.78
1508  2013          Action   2078618.23
...    ...             ...          ...
1647  2013  Web Publishing      2321.34
390   2005  Web Publishing      1564.19
2270  2014  Web Publishing       372.25
5221  2017  Web Publishing       156.37
4744  2016  Web Publishing         0.02

[320 rows x 3 columns]

### 2. UserForGenre
Debe devolver el usuario que acumula más horas jugadas para el género dado y una lista de la acumulación de horas jugadas por año.

Ejemplo de retorno: {"Usuario con más horas jugadas para Género X" : us213ndjss09sdf, "Horas jugadas":[{Año: 2013, Horas: 203}, {Año: 2012, Horas: 100}, {Año: 2011, Horas: 23}]}

In [ ]:
#Se eliminan las columnas que no son necesarias 
df_steam_games = df_steam_games.drop(['title', 'developer', 'price', 'tags','specs'],axis=1)

In [ ]:
# Join de los dataframes
join_items_games=pd.merge(df_user_items,df_steam_games,left_on='item_id', right_on='item_id', how='right')

join_items_games.drop(['item_id'], axis=1, inplace=True)
join_items_games.dropna(inplace=True)

join_items_games['hours_game'] = join_items_games['hours_game'].astype(int)
join_items_games['year'] = join_items_games['year'].astype(int)

In [ ]:
#este código fragmenta una columna de listas en varias filas para hacer que los datos sean más accesibles y manejables.
join_items_games['genres'] = join_items_games['genres'].apply(ast.literal_eval)

In [ ]:
#Expandimos las listas en filas
join_items_games = join_items_games.explode('genres').reset_index(drop=True)

In [ ]:
#Encontrar el jugador con más horas jugadas por género y año
idxmax_genre_user = join_items_games .loc[join_items_games .groupby(['genres', 'year'])['hours_game'].idxmax()]

In [ ]:
#Filtrar las columnas necesarias
idxmax_genre_user = idxmax_genre_user[['genres', 'year', 'user_id', 'hours_game']]

In [ ]:
#Segunda agrupación para obtener las horas jugadas por año para el usuario seleccionado
max_hours_genre_user = idxmax_genre_user.groupby(['genres', 'year', 'user_id'])['hours_game'].sum().reset_index()

In [ ]:
#max_hours_genre_user.to_csv('data/DataAPI/UserForGenre.csv', index=False)

### 3. UsersRecommend
Devuelve el top 3 de juegos MÁS recomendados por usuarios para el año dado. (reviews.recommend = True y comentarios positivos/neutrales)

Ejemplo de retorno: [{"Puesto 1" : X}, {"Puesto 2" : Y},{"Puesto 3" : Z}]

In [61]:
# JOIN de los dataframes
join_reviews_games = pd.merge(df_user_reviews, df_steam_games, on='item_id', how='left')

In [62]:
join_reviews_games["year_posted"].astype('Int64')

0        2011
1        2013
2        2014
3        2014
4        2014
         ... 
45750    2015
45751    2015
45752    2011
45753    2014
45754    2015
Name: year_posted, Length: 45755, dtype: Int64

In [63]:
join_reviews_games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45755 entries, 0 to 45754
Data columns (total 13 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   user_id             45755 non-null  object 
 1   funny               6610 non-null   object 
 2   item_id             45755 non-null  int64  
 3   recommend           45755 non-null  bool   
 4   year_posted         43625 non-null  float64
 5   sentiment_analysis  45755 non-null  int64  
 6   genres              38047 non-null  object 
 7   title               38047 non-null  object 
 8   tags                38047 non-null  object 
 9   specs               38047 non-null  object 
 10  price               38047 non-null  float64
 11  developer           38047 non-null  object 
 12  year                37865 non-null  float64
dtypes: bool(1), float64(3), int64(2), object(7)
memory usage: 4.2+ MB


In [64]:
# Filtrar las filas que cumplen con las condiciones
filter = join_reviews_games.loc[(join_reviews_games['recommend'] == True) & (join_reviews_games['sentiment_analysis'].isin([1, 2])),
                           ['year_posted', 'title']]

In [65]:
# Renombrar la columna 'year_posted' a 'year'
filter = filter.rename(columns={'year_posted': 'year'})

# Agrupar por 'year' y 'name' y contar las ocurrencias
grouped_result = filter.groupby(['year', 'title']).size().reset_index(name='count')

# Ordenar por 'year' y 'count' en orden descendente
grouped_result = grouped_result.sort_values(by=['year', 'count'], ascending=[False, False])

# Obtener el top 3 por año
top3_x_year = grouped_result.groupby('year').head(3)

In [68]:
top3_x_year["year"] = top3_x_year["year"].astype('Int64')

C:\Users\FRANK\AppData\Local\Temp\ipykernel_19788\895433307.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  top3_x_year["year"] = top3_x_year["year"].astype('Int64')


In [70]:
top3_x_year

year                             title  count
2121  2015  Counter-Strike: Global Offensive   1195
2964  2015                   Team Fortress 2    560
2351  2015                       Garry's Mod    326
1687  2014                   Team Fortress 2    904
1005  2014  Counter-Strike: Global Offensive    655
1202  2014                       Garry's Mod    477
742   2013                   Team Fortress 2    463
519   2013                       Garry's Mod    229
580   2013                     Left 4 Dead 2    115
307   2012                   Team Fortress 2    182
226   2012                       Garry's Mod     27
309   2012                          Terraria     27
143   2011                   Team Fortress 2     58
119   2011                          Portal 2     20
144   2011                          Terraria     16
24    2010                   Team Fortress 2      6
0     2010                       Alien Swarm      4
12    2010                     Killing Floor      4

In [71]:
# Exportar el resultado en un archivo CSV
top3_x_year.to_csv(r'C:\PI1\data\DataAPI\UsersRecommend.csv', index=False)

### 4. UsersNotRecommend
Devuelve el top 3 de juegos MENOS recomendados por usuarios para el año dado. (reviews.recommend = False y comentarios negativos)

Ejemplo de retorno: [{"Puesto 1" : X}, {"Puesto 2" : Y},{"Puesto 3" : Z}]

In [ ]:
# LEFT JOIN a dataframes
join_not_recom = pd.merge(df_user_reviews, df_steam_games, on='item_id', how='left')

In [ ]:
join_not_recom

user_id funny  item_id  recommend  year_posted  \
0          76561198040188061   NaN       10       True         2011   
1                  epic_doom   NaN       10       True         2013   
2             mayshowganmore   NaN       10       True         2014   
3      BestinTheWorldThund3r   NaN       10       True         2014   
4          76561198072207162   NaN       10       True         2014   
...                      ...   ...      ...        ...          ...   
45750      76561198068265347   NaN    99900       True         2015   
45751      76561198073638107   NaN    99900      False         2015   
45752              Gatsukama   NaN    99910       True         2011   
45753             Themfgamer   NaN    99910      False         2014   
45754               Jezza453   NaN    99910       True         2015   

       sentiment_analysis                                             genres  \
0                       2                                         ['Action']   
1                       1                                         ['Action']   
2                       2                                         ['Action']   
3                       2                                         ['Action']   
4                       2                                         ['Action']   
...                   ...                                                ...   
45750                   2                                                NaN   
45751                   2                                                NaN   
45752                   2  ['Adventure', 'Casual', 'Free to Play', 'Massi...   
45753                   0  ['Adventure', 'Casual', 'Free to Play', 'Massi...   
45754                   2  ['Adventure', 'Casual', 'Free to Play', 'Massi...   

                title                                               tags  \
0      Counter-Strike  ['Action', 'FPS', 'Multiplayer', 'Shooter', 'C...   
1      Counter-Strike  ['Action', 'FPS', 'Multiplayer', 'Shooter', 'C...   
2      Counter-Strike  ['Action', 'FPS', 'Multiplayer', 'Shooter', 'C...   
3      Counter-Strike  ['Action', 'FPS', 'Multiplayer', 'Shooter', 'C...   
4      Counter-Strike  ['Action', 'FPS', 'Multiplayer', 'Shooter', 'C...   
...               ...                                                ...   
45750             NaN                                                NaN   
45751             NaN                                                NaN   
45752  Puzzle Pirates  ['Free to Play', 'Massively Multiplayer', 'Puz...   
45753  Puzzle Pirates  ['Free to Play', 'Massively Multiplayer', 'Puz...   
45754  Puzzle Pirates  ['Free to Play', 'Massively Multiplayer', 'Puz...   

                                                   specs  price    developer  \
0           ['Multi-player', 'Valve Anti-Cheat enabled']   9.99        Valve   
1           ['Multi-player', 'Valve Anti-Cheat enabled']   9.99        Valve   
2           ['Multi-player', 'Valve Anti-Cheat enabled']   9.99        Valve   
3           ['Multi-player', 'Valve Anti-Cheat enabled']   9.99        Valve   
4           ['Multi-player', 'Valve Anti-Cheat enabled']   9.99        Valve   
...                                                  ...    ...          ...   
45750                                                NaN    NaN          NaN   
45751                                                NaN    NaN          NaN   
45752  ['Single-player', 'Multi-player', 'MMO', 'Co-o...   0.00  Three Rings   
45753  ['Single-player', 'Multi-player', 'MMO', 'Co-o...   0.00  Three Rings   
45754  ['Single-player', 'Multi-player', 'MMO', 'Co-o...   0.00  Three Rings   

       year  
0      2000  
1      2000  
2      2000  
3      2000  
4      2000  
...     ...  
45750  <NA>  
45751  <NA>  
45752  2011  
45753  2011  
45754  2011  

[45755 rows x 13 columns]

In [ ]:
# Filtrar las filas que cumplen con las condiciones
df_filtered = join_not_recom.loc[(join_not_recom['recommend'] == False) & (join_not_recom['sentiment_analysis'] == 0),
                           ['year', 'title']]

In [ ]:
df_filtered 

year                             title
93     2009  Call of Duty®: Modern Warfare® 2
99     2009  Call of Duty®: Modern Warfare® 2
104    2009  Call of Duty®: Modern Warfare® 2
132    2009  Call of Duty®: Modern Warfare® 2
148    2009  Call of Duty®: Modern Warfare® 2
...     ...                               ...
45700  <NA>                               NaN
45702  <NA>                               NaN
45733  <NA>                               NaN
45748  <NA>                               NaN
45753  2011                    Puzzle Pirates

[2596 rows x 2 columns]

In [ ]:
# Contar las ocurrencias de cada desarrolladora por año
developer_counts = df_filtered.groupby(['year', 'title']).size().reset_index(name='count')

In [ ]:
developer_counts

year                          title  count
0    1990                 Commander Keen      1
1    1998   YOU DON'T KNOW JACK HEADRUSH      1
2    1999        Omikron: The Nomad Soul      1
3    1999  RollerCoaster Tycoon®: Deluxe      1
4    1999                     Sven Co-op      1
..    ...                            ...    ...
691  2017                    Gang Beasts      2
692  2017                      Robocraft     34
693  2017      Skara - The Blade Remains      1
694  2017                   Tree of Life      1
695  2017                       Unturned     39

[696 rows x 3 columns]

In [ ]:
# Ordenar por 'year' y 'count' en orden descendente
grouped_result = developer_counts.sort_values(by=['year', 'count'], ascending=[False, False]) 

In [ ]:
grouped_result

year                          title  count
695  2017                       Unturned     39
685  2017          ARK: Survival Evolved     34
692  2017                      Robocraft     34
688  2017                  Codename CURE      4
687  2017                     Brawlhalla      3
..    ...                            ...    ...
2    1999        Omikron: The Nomad Soul      1
3    1999  RollerCoaster Tycoon®: Deluxe      1
4    1999                     Sven Co-op      1
1    1998   YOU DON'T KNOW JACK HEADRUSH      1
0    1990                 Commander Keen      1

[696 rows x 3 columns]

In [ ]:
# Obtener el top 3 por año
result_df = grouped_result.groupby('year').head(3)  

In [ ]:
# exportar el dataframe en un archivo CSV
#result_df.to_csv('data/DataAPI/UsersNotRecommend.csv', index=False)

### 5. sentiment_analysis
Según el año de lanzamiento, se devuelve una lista con la cantidad de registros de reseñas de usuarios que se encuentren categorizados con un análisis de sentimiento.

Ejemplo de retorno: {Negative = 182, Neutral = 120, Positive = 278}

In [43]:
# INNER JOIN de los DataFrames
join_game_review_sent = pd.merge(df_steam_games, df_user_reviews, on='item_id', how='inner')

In [44]:
# Agrupar por desarrolladora y sentimiento, y contar las ocurrencias
join_game_review_sent = join_game_review_sent.groupby(["year", 'sentiment_analysis']).size().reset_index(name='count')

In [45]:
join_game_review_sent

year  sentiment_analysis  count
0   1989.0                   2      1
1   1990.0                   0      1
2   1990.0                   2      4
3   1991.0                   2      1
4   1992.0                   0      1
..     ...                 ...    ...
71  2016.0                   1    235
72  2016.0                   2   1746
73  2017.0                   0    299
74  2017.0                   1    139
75  2017.0                   2   1126

[76 rows x 3 columns]

In [46]:
# Cambio de orden entre columnas
join_game_review_sent= join_game_review_sent.pivot(index='year', columns='sentiment_analysis', values='count').reset_index()

In [47]:
join_game_review_sent

sentiment_analysis    year       0      1       2
0                   1989.0     NaN    NaN     1.0
1                   1990.0     1.0    NaN     4.0
2                   1991.0     NaN    NaN     1.0
3                   1992.0     1.0    NaN     2.0
4                   1993.0     1.0    NaN     3.0
5                   1994.0     3.0    3.0     9.0
6                   1995.0     5.0    NaN    10.0
7                   1996.0     NaN    3.0     4.0
8                   1997.0     8.0    3.0    31.0
9                   1998.0    12.0    9.0    57.0
10                  1999.0    12.0    4.0    35.0
11                  2000.0     9.0   10.0    50.0
12                  2001.0    10.0    NaN    21.0
13                  2002.0     7.0    NaN    32.0
14                  2003.0    47.0   25.0   109.0
15                  2004.0    60.0   58.0   294.0
16                  2005.0    31.0   24.0   149.0
17                  2006.0   214.0  201.0  1115.0
18                  2007.0   315.0  322.0  2175.0
19                  2008.0    47.0   30.0   232.0
20                  2009.0   287.0  151.0   988.0
21                  2010.0   270.0  152.0  1105.0
22                  2011.0   390.0  230.0  1827.0
23                  2012.0  1243.0  715.0  4679.0
24                  2013.0  1043.0  564.0  4206.0
25                  2014.0  1059.0  483.0  3512.0
26                  2015.0  1072.0  550.0  3483.0
27                  2016.0   502.0  235.0  1746.0
28                  2017.0   299.0  139.0  1126.0

In [48]:
# Renombrar las columnas según el formato deseado
join_game_review_sent.columns = ['Year', 'Negative', 'Neutral', 'Positive']

In [49]:
join_game_review_sent

Year  Negative  Neutral  Positive
0   1989.0       NaN      NaN       1.0
1   1990.0       1.0      NaN       4.0
2   1991.0       NaN      NaN       1.0
3   1992.0       1.0      NaN       2.0
4   1993.0       1.0      NaN       3.0
5   1994.0       3.0      3.0       9.0
6   1995.0       5.0      NaN      10.0
7   1996.0       NaN      3.0       4.0
8   1997.0       8.0      3.0      31.0
9   1998.0      12.0      9.0      57.0
10  1999.0      12.0      4.0      35.0
11  2000.0       9.0     10.0      50.0
12  2001.0      10.0      NaN      21.0
13  2002.0       7.0      NaN      32.0
14  2003.0      47.0     25.0     109.0
15  2004.0      60.0     58.0     294.0
16  2005.0      31.0     24.0     149.0
17  2006.0     214.0    201.0    1115.0
18  2007.0     315.0    322.0    2175.0
19  2008.0      47.0     30.0     232.0
20  2009.0     287.0    151.0     988.0
21  2010.0     270.0    152.0    1105.0
22  2011.0     390.0    230.0    1827.0
23  2012.0    1243.0    715.0    4679.0
24  2013.0    1043.0    564.0    4206.0
25  2014.0    1059.0    483.0    3512.0
26  2015.0    1072.0    550.0    3483.0
27  2016.0     502.0    235.0    1746.0
28  2017.0     299.0    139.0    1126.0

In [50]:
# Rellenar NaN con 0
join_game_review_sent = join_game_review_sent.fillna(0)

In [55]:
join_game_review_sent

Year  Negative  Neutral  Positive
0   1989.0         0        0         1
1   1990.0         1        0         4
2   1991.0         0        0         1
3   1992.0         1        0         2
4   1993.0         1        0         3
5   1994.0         3        3         9
6   1995.0         5        0        10
7   1996.0         0        3         4
8   1997.0         8        3        31
9   1998.0        12        9        57
10  1999.0        12        4        35
11  2000.0         9       10        50
12  2001.0        10        0        21
13  2002.0         7        0        32
14  2003.0        47       25       109
15  2004.0        60       58       294
16  2005.0        31       24       149
17  2006.0       214      201      1115
18  2007.0       315      322      2175
19  2008.0        47       30       232
20  2009.0       287      151       988
21  2010.0       270      152      1105
22  2011.0       390      230      1827
23  2012.0      1243      715      4679
24  2013.0      1043      564      4206
25  2014.0      1059      483      3512
26  2015.0      1072      550      3483
27  2016.0       502      235      1746
28  2017.0       299      139      1126

In [57]:
# Convertir las columnas a tipo entero
join_game_review_sent[['Year','Negative', 'Neutral', 'Positive']] = join_game_review_sent[['Year','Negative', 'Neutral', 'Positive']].astype(int)

In [58]:
join_game_review_sent

Year  Negative  Neutral  Positive
0   1989         0        0         1
1   1990         1        0         4
2   1991         0        0         1
3   1992         1        0         2
4   1993         1        0         3
5   1994         3        3         9
6   1995         5        0        10
7   1996         0        3         4
8   1997         8        3        31
9   1998        12        9        57
10  1999        12        4        35
11  2000         9       10        50
12  2001        10        0        21
13  2002         7        0        32
14  2003        47       25       109
15  2004        60       58       294
16  2005        31       24       149
17  2006       214      201      1115
18  2007       315      322      2175
19  2008        47       30       232
20  2009       287      151       988
21  2010       270      152      1105
22  2011       390      230      1827
23  2012      1243      715      4679
24  2013      1043      564      4206
25  2014      1059      483      3512
26  2015      1072      550      3483
27  2016       502      235      1746
28  2017       299      139      1126

In [60]:
# Guardar el resultado en un archivo CSV
join_game_review_sent.to_csv('C:/PI1/data/DataAPI/sentiment_analysis.csv', index=False)